File: 02-Sorting/yeonsoo/K번째수.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    int size = commands.size();
    
    for(int i{0}; i<size; i++){
        vector<int> sliced_array (array.begin()+commands[i][0]-1, array.begin()+commands[i][1]); // 하나 앞까지 자름
        sort(sliced_array.begin(), sliced_array.end());
        answer.push_back(sliced_array[commands[i][2]-1]);
    }
    return answer;
}

File: 02-Sorting/yeonsoo/K번째수.py
def solution(array, commands):
    answer = []
    size_ = len(commands)
    for command in commands:
        a,b,c = command
        answer.append((sorted(array[a-1:b]))[c-1])
    return answer

# def solution(array, commands):
#     return list(map(lambda x:sorted(array[x[0]-1:x[1]])[x[2]-1], commands))
